### PR TITLE
#168781646: Add accommodation

### PR DESCRIPTION
**Pull request by alainmateso**
#### What does this PR do?
Create new  accommodation facilities

#### Description of Task to be completed?
suppliers and travel admins can create accommodations

#### How should this be manually tested?
Clone this branch
run `npm i`
run `npm run dev`
navigate to `http://localhost:<:port>/accommodations/new`
fill the form with valid data
Click create

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#168781646

#### Screenshots (if appropriate)
Screen Shot 2019-12-12 at 22 03 54

#### Questions:
N/A

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -1,19 +1,34 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+// import { Container } from 'react-bootstrap';
```
Comment (sabin18) [inline src/components/Home.js]: remove this comment

**Review by sabin18 — CHANGES_REQUESTED** [review]
good job @alainmateso, work on the comment.

**Review by sabin18 — APPROVED** [review]

**Review by ngireric123 — APPROVED** [review]
LGTM

File: src/actions/accommodationActions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { ADD_ACCOMMODATION_SUCESS, ADD_ACCOMMODATION_FAILURE } from './types';
+import backendCall from '../helpers/backendCall';
+import authHelper from '../helpers/authHelper';
+
+const { getToken } = authHelper;
+
+const operationType = (type, payload) => ({
```
Comment (nakiwuge) [inline src/actions/accommodationActions.js]: rename this to `accommodationType`

File: src/actions/locationActions.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { GET_LOCATIONS, GET_LOCATIONS_ERROR } from './types';
+import authHelper from '../helpers/authHelper';
+import backendCall from '../helpers/backendCall';
+
+const { getToken } = authHelper;
+
+const token = getToken();
+
+const authType = (type, payload) => ({
```
Comment (nakiwuge) [inline src/actions/locationActions.js]: rename this to `locationsType`

File: src/components/pages/CreateAccommodation.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    static getDerivedStateFromProps(nextProps) {
+      const {
+        accommodationData, status,
+      } = nextProps;
+
+      switch (status) {
+        case 'Success':
+          window.location.replace(`/accommodations/${accommodationData.data.slug}`);
```
Comment (nakiwuge) [inline src/components/pages/CreateAccommodation.js]: Make use of react by using  `history.push`

File: src/views/accommodations/NewAccommodation.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import CreateAccommodation from '../../components/pages/CreateAccommodation';
+
+export const Accommodations = () => (
+      <CreateAccommodation />
+
```
Comment (nakiwuge) [inline src/views/accommodations/NewAccommodation.js]: remove this line

File: src/views/accommodations/AllAccommodations.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+                <h1>All Accommodations</h1>
+                { checkSupplier() ? (
+                    <ButtonToolbar>
+                        <Button href="/accommodations/new">
+                            <Add />
+                            Create new accommodation
+                        </Button>
+                    </ButtonToolbar>
+                ) : null}
+
```
Comment (nakiwuge) [inline src/views/accommodations/AllAccommodations.js]: remove this line

**Pull request merged**